tests/fixtures.py

```diff
@@ -144,3 +144,35 @@ def create_rbd_pod(request):
         interface_type=constants.CEPHBLOCKPOOL,
         pvc_name=class_instance.pvc_obj.name
     )
+
+
+@pytest.fixture()
+def create_project(request):
+    """
+    Create a project
+    """
+    class_instance = request.node.cls
+
+    def finalizer():
+        """
+        Delete the project
+        """
+        class_instance.project_obj.delete(
+            resource_name=class_instance.namespace
+        )
+        class_instance.project_obj.wait_for_delete(
+            resource_name=class_instance.namespace
+        )
+
+    request.addfinalizer(finalizer)
+
+    class_instance.namespace = helpers.create_unique_resource_name(
+        'test', 'namespace'
+    )
+    class_instance.project_obj = ocp.OCP(
+        kind='Project', namespace=class_instance.namespace
+    )
+
+    assert class_instance.project_obj.new_project(class_instance.namespace), (
+        f'Failed to create new project {class_instance.namespace}'
+    )
```

Comment:
This fixture is pushing namespace object into a class, which is error prone hack. As noted in [Coding guidelines for OCS-CI project](https://github.com/red-hat-storage/ocs-ci/blob/master/docs/coding_guidelines.md), we should not use hacks like that:

> Code Readability takes priority, Lets avoid crazy hacks and tricks and stick to straight forward code blocks that are easy to understand for all.

Reply:
As this is still under discussion and not a hack in any way, let's not block this PR based on that

Reply:
> As this is ... not a hack in any way

It's up to you if you don't want to block this PR on this problem after all. You can argue that making an exception here makes sense for some objective reason, without any need to refuse to see the problems behind this design. Saying that this is not a hack in any way is little problematic.

Reply:
This fixture is already merged #402